File: src/controllers/recipeController.ts
import { FastifyRequest, FastifyReply } from 'fastify';
import { getRecipes } from '../storage';
import { createRecipe } from '../services/recipeService';
import { Recipe } from '../entities/Recipe';

export const getAllRecipes = async (request: FastifyRequest, reply: FastifyReply) => {
  const recipes = getRecipes();
  return reply.send(recipes);
};

export const createNewRecipe = async (request: FastifyRequest, reply: FastifyReply) => {
  const recipeData = request.body as Partial<Recipe>;
  const recipe = createRecipe(recipeData);
  return reply.status(201).send(recipe);
};

// ... similar changes for other endpoint handlers ...

File: app/controllers/recipe_controller.py
from fastapi import APIRouter, Request, HTTPException
from typing import List, Dict
from app.models.recipe import Recipe
from app.services.recipe_service import create_recipe
from app.storage import get_recipes

router = APIRouter()

@router.get("/recipes", response_model=List[Recipe])
async def get_all_recipes():
    return get_recipes()

@router.post("/recipes", response_model=Recipe, status_code=201)
async def create_new_recipe(recipe_data: Dict):
    return create_recipe(recipe_data)
